File: drestaurant-libs/drestaurant-courier/src/main/java/com/drestaurant/courier/domain/CourierCommandHandler.java
package com.drestaurant.courier.domain;

import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;

import org.axonframework.commandhandling.CommandHandler;
import org.axonframework.commandhandling.model.Aggregate;
import org.axonframework.commandhandling.model.AggregateNotFoundException;
import org.axonframework.commandhandling.model.Repository;
import org.axonframework.eventhandling.EventBus;

public class CourierCommandHandler {

	private Repository<Courier> repository;
	private EventBus eventBus;

	public CourierCommandHandler(Repository<Courier> repository, EventBus eventBus) {
		this.repository = repository;
		this.eventBus = eventBus;
	}

	@CommandHandler
	public void handle(ValidateOrderByCourierCommand command) {
		try {
			Aggregate<Courier> courierAggregate = repository.load(command.getCourierId());
			courierAggregate.execute(courier -> courier.validateOrder(command.getOrderId(), command.getAuditEntry()));
		} catch (AggregateNotFoundException exception) {
			eventBus.publish(asEventMessage(new CourierNotFoundForOrderEvent(command.getCourierId(), command.getOrderId(), command.getAuditEntry())));
		}
	}

}

File: drestaurant-libs/drestaurant-courier/src/main/kotlin/com/drestaurant/courier/domain/CourierCommandHandler.kt
package com.drestaurant.courier.domain

import org.axonframework.commandhandling.CommandHandler
import org.axonframework.eventhandling.EventBus
import org.axonframework.eventhandling.GenericEventMessage.asEventMessage
import org.axonframework.modelling.command.AggregateNotFoundException
import org.axonframework.modelling.command.Repository

internal open class CourierCommandHandler(private val repository: Repository<Courier>, private val eventBus: EventBus) {

    @CommandHandler
    fun handle(command: ValidateOrderByCourierInternalCommand) = try {
        repository.load(command.courierId.identifier).execute { it.validateOrder(command.targetAggregateIdentifier, command.auditEntry) }
    } catch (exception: AggregateNotFoundException) {
        eventBus.publish(asEventMessage<Any>(CourierNotFoundForOrderInternalEvent(command.courierId, command.targetAggregateIdentifier, command.auditEntry)))
    }
}
